encoding/protoavro/marshal.go

```diff
@@ -10,10 +10,42 @@ import (
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
+type MarshalOptions struct {
+	ExtraFields []ExtraField
+}
+
+type ExtraField struct {
+	FieldName string
+	Message proto.Message
+}
```

Comment:
Can `MarshalOptions` be used for both `Marshal` and `Unmarshal` (perhaps with a better name in that case)?